### PR TITLE
Support channel setting: AutomaticallyCompleteFullyPaidCheckouts

### PR DESCRIPTION
**Pull request by lkostrowski**
Waiting for API to be released everywhere in Saleor Cloud

## What type of PR is this?
- [ ] 💅 Refactor
- [x] 🌟 Feature
- [ ] 🔥 Bug Fix
- [ ] 🔩 Maintenance
- [ ] 🛠 Workflow CI/CD changes

## Related Issues or Documents

https://github.com/saleor/saleor-docs/pull/1334
https://github.com/saleor/saleor/pull/16782

- closes #

## Usage Instructions, Screenshots, Recordings

CleanShot 2024-10-04 at 09 41 42

## Have you written tests?
- [ ] Yes!
- [x] No... here is why: I don't see what to test here

### CODE REVIEW

File: src/channels/components/ChannelForm/AutomaticallyCompleteCheckouts.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    </Checkbox>
+    <Box paddingLeft={4}>
+      {" "}
+      <Text size={3} color="default2" paddingLeft={0.5}>
+        <FormattedMessage
+          {...messages.automaticallyCompleteCheckoutsDescription}
+          values={{
+            link: (
+              <Link
+                href="https://docs.saleor.io/developer/payments/transactions#automatic-checkout-completion"
```
Comment (poulch) [inline src/channels/components/ChannelForm/AutomaticallyCompleteCheckouts.tsx:45]: question: what about putting this link in `src/links.ts`?

File: src/channels/components/ChannelForm/ChannelForm.tsx [hunk 2]
```diff
@@ -212,6 +216,13 @@ export const ChannelForm: React.FC<ChannelFormProps> = ({
           hasError={!!formErrors.defaultTransactionFlowStrategy}
           disabled={disabled}
         />
+        <Box />
```
Comment (poulch) [inline src/channels/components/ChannelForm/ChannelForm.tsx:219]: questtion: do we need this empty box?

File: src/channels/pages/ChannelDetailsPage/ChannelDetailsPage.tsx [hunk 3]
```diff
@@ -211,6 +218,13 @@ const ChannelDetailsPage = function <TErrors extends ChannelErrorFragment[]>({
                 : TransactionFlowStrategyEnum.CHARGE,
           });
         };
+
+        const handleAutomaticallyCompleteCheckoutsChange = () => {
+          set({
+            automaticallyCompleteCheckouts: !data.automaticallyCompleteCheckouts,
```
Comment (poulch) [inline src/channels/pages/ChannelDetailsPage/ChannelDetailsPage.tsx:224]: question: why do you use `!`?

**Review by poulch — APPROVED** [review]

**Comment by szczecha** [comment]
1. Empty order history for the event `PLACED_AUTOMATICALLY_FROM_PAID_CHECKOUT`  Screenshot 2024-10-04 at 09 56 13
2. When creating a new channel, we don't send checkout settings in `channelCreate` input

https://github.com/user-attachments/assets/1b12b71c-acc7-4388-8134-a2860c2714c2

File: src/channels/components/ChannelForm/ChannelForm.tsx (hunk 2 above)
Comment (lkostrowski) [reply src/channels/components/ChannelForm/ChannelForm.tsx:219]: CleanShot 2024-10-04 at 11 03 49

container is written incorrectly, so Box seems to be used to push the component to the new line. This should be refactored but its out of the scope of my changes

File: src/channels/pages/ChannelDetailsPage/ChannelDetailsPage.tsx (hunk 3 above)
Comment (lkostrowski) [reply src/channels/pages/ChannelDetailsPage/ChannelDetailsPage.tsx:224]: Becasue its a checkbox, so it inverts data (false → true, true → false)
The handler doesnt receive the change, it uses current data to flip the state

File: src/channels/components/ChannelForm/AutomaticallyCompleteCheckouts.tsx (hunk 1 above)
Comment (lkostrowski) [reply src/channels/components/ChannelForm/AutomaticallyCompleteCheckouts.tsx:45]: Good idea. I suggest you add ESLint rule that will suggest such change

**Comment by lkostrowski** [comment]
CleanShot 2024-10-04 at 11 17 47

**Comment by lkostrowski** [comment]
@szczecha @poulch applied fixes for your comments

**Review by szczecha — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Pull request merged**